src/tools/tidy/src/target_specific_tests.rs

```diff
@@ -53,9 +53,9 @@ pub fn check(path: &Path, bad: &mut bool) {
             } else if directive.starts_with(COMPILE_FLAGS_HEADER) {
                 let compile_flags = &directive[COMPILE_FLAGS_HEADER.len()..];
                 if let Some((_, v)) = compile_flags.split_once("--target") {
-                    if let Some((arch, _)) =
-                        v.trim_start_matches(|c| c == ' ' || c == '=').split_once("-")
-                    {
+                    let v = v.trim_start_matches(|c| c == ' ' || c == '=');
+                    let v = if v == "{{target}}" { Some((v, v)) } else { v.split_once("-") };
+                    if let Some((arch, _)) = v {
```

Comment:
This just puts `{{target}}` into `arch`, probably not ideal but currently the arch string is not used anyways so it's unclear what would be the best thing to do here for now until things do get validated.

Reply:
Did you consider adding a new field to `RevisionInfo`? Since `target_arch` is an `Option`, we can leave it as `None` for `{{target}}` values.